lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -354,25 +355,24 @@ internal void RecordRunningTaskDuringSystemFailure(IRunningTask runningTask, str
         }
 
         /// <summary>
-        /// Gets error type based on the information in IFailedTask 
-        /// Currently we use the Message in IFailedTask to distinguish different types of errors
+        /// Gets error type based on the exception type in IFailedTask 
         /// </summary>
         /// <param name="failedTask"></param>
         /// <returns></returns>
-        private TaskStateEvent GetTaskErrorEvent(IFailedTask failedTask)
+        private TaskStateEvent GetTaskErrorEventByExceptionType(IFailedTask failedTask)
         {
-            switch (failedTask.Message)
+            if (failedTask.AsError() is IMRUTaskAppException)
             {
-                case TaskAppError:
-                    _numberOfAppErrors++;
-                    return TaskStateEvent.FailedTaskAppError;
-                case TaskSystemError:
-                    return TaskStateEvent.FailedTaskSystemError;
-                case TaskGroupCommunicationError:
-                    return TaskStateEvent.FailedTaskCommunicationError;
-                default:
-                    return TaskStateEvent.FailedTaskSystemError;
+                _numberOfAppErrors++;
+                return TaskStateEvent.FailedTaskAppError;
             }
+
+            if (failedTask.AsError() is IMRUTaskGroupCommunicationException)
+            {
+                return TaskStateEvent.FailedTaskCommunicationError;
+            }
+
+            return TaskStateEvent.FailedTaskSystemError;
```

Comment:
This method is used to get exception type so that to move to the next task failed state. Those failed stated will be treated same in fault tolerant.